plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -194,13 +194,13 @@ public void outputFailedPlugins() {
      * @param plugins list of plugins to download
      */
     public void downloadPlugins(List<Plugin> plugins) {
-        for (Plugin plugin : plugins) {
+        plugins.parallelStream().forEach(plugin -> {
```

Comment:
can you run it on your setup and see if it helps 😄

Reply:
That was the plan! I am just abusing Jenkins CI to get the jar :)

Reply:
Holy crap 😓
```
real    1m 6.93s
user    0m 8.98s
sys     0m 1.86s
```

I think we can optimize it even more if we play with custom thread pool and just doing it on all available cores (+hyper threading)

Reply:
but reading up on optimizing I would say NOPE 😅